src/main/java/io/aiven/kafka/connect/http/sender/OAuth2HttpSender.java

```diff
@@ -64,14 +66,14 @@ protected HttpResponse<String> sendWithRetries(final HttpRequest.Builder request
         return super.sendWithRetries(accessTokenAwareRequestBuilder, handler);
     }
 
-    private HttpRequest.Builder loadOrRefreshAccessToken(final HttpRequest.Builder requestBuilder) {
+    private HttpRequest.Builder loadOrRefreshAccessToken(final HttpRequest.Builder requestBuilder, HttpRequestBuilder accessTokenRequestBuilder ) {
```

Comment:
you don't need to pass the builder in the parameters as you already have it your class instance